fecgov/openFEC
# Create new end point to calculate total of independent expenditures by candidate

## Summary (required)

- Resolves #4061 

New endpoint is created to calculate total independent expenditures grouped by candidate by cycle by support_oppose_indicator. The following three filters are available to the endpoint. None of them is mandatory.

election_full: true by default
candidate_id
cycle ( mapping to different columns depending on the value of election_full)

Totals of different support_oppose_indicator per cand_id per cycle are returned.

## How to test the changes locally
- Download feature branch
- Run pytest to make sure all tests pass
- Run api on local with setting SQLA_CONN to DEV database
- Returning total independent expenditure for P00003392 on candidate_election_year 2008 with support_oppose_indicator
http://127.0.0.1:5000/v1/schedules/schedule_e/totals/by_candidate/?cycle=2008&per_page=20&candidate_id=P00003392&sort_null_only=false&election_full=true&sort_nulls_last=false&page=1&sort_hide_null=false
- Returning total independent expenditure for P80001571 per fec_cyce
http://127.0.0.1:5000/v1/schedules/schedule_e/totals/by_candidate/?per_page=20&candidate_id=P80001571&sort_null_only=false&election_full=false&sort_nulls_last=false&page=1&sort_hide_null=false
- Returning total independent expenditure on each candidate per cand_election_year
http://127.0.0.1:5000/v1/schedules/schedule_e/totals/by_candidate/?per_page=20&sort_null_only=false&election_full=true&sort_nulls_last=false&page=1&sort_hide_null=false

## Impacted areas of the application
List general components of the application that this PR will affect:

-  This is a new endpoint under TAG "independent expenditures"
/schedules/schedule_e/totals/by_candidate/

No existing API will be impacted.

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
fix/other_pr | [link]()
feature/other_pr | [link]()
